### PR TITLE
Added git redirects back to the router

### PR DESCRIPTION
**Pull request by skeswa**
It broke some tests, but we'll take care of that in the goconvey migration PR.

### CODE REVIEW

File: common/depot/build_hashed_repo_name.go [hunk 1]
```diff
@@ -9,10 +9,6 @@ const (
 	// DepotInternalServiceAddress is the address for which all internal requests
 	// will be resolved.
 	DepotInternalServiceAddress = "depot-int-svc"
-	// externalRepoURLTemplate is the URL of a repo in depot that is accessible
```
Comment (Shikkic) [inline common/depot/build_hashed_repo_name.go]: Is this just clean up that wasn't necessary from the last depot PR?

Comment (skeswa) [inline common/depot/build_hashed_repo_name.go]: yeeppppp

File: router/package_request.go [hunk 2]
```diff
@@ -59,7 +62,7 @@ func newPackageRequest(args newPackageRequestArgs) (*packageRequest, error) {
 
 	// Only go out to github to get the matched SHA if the matched SHA is
 	// necessary.
-	if isGoGetRequest(args.req) {
+	if isGoGetRequest(args.req) || isGitRequest(parts) {
```
Comment (Shikkic) [inline router/package_request.go]: I thought we agree'd we were getting rid of the the `isGoGetRequest` check here? I've removed it in my PR

File: router/package_request.go [hunk 3]
```diff
@@ -126,6 +129,40 @@ type respondToPackageRequestArgs struct {
...
+			pr.parts.subpath)
+
+		// Issue a permanent redirect.
+		http.Redirect(
+			args.res,
+			pr.req,
+			redirectLocation,
+			http.StatusMovedPermanently)
+
+		// Without blocking, count a packfile request as a package download.
```
Comment (Shikkic) [inline router/package_request.go]: By `count a packfile request as a package download.` are talking about clone requests coming from the git client?

File: router/package_request.go (hunk 2 above)
Comment (skeswa) [inline router/package_request.go]: Yeah. Would have been a mess to change it in both places. This is a cleaner merge bc its exactly one changed line.

**Comment by Shikkic** [comment]
Makes sense to me.

File: router/package_request.go (hunk 3 above)
Comment (skeswa) [inline router/package_request.go]: yeahman

**Pull request merged**